src/app/azure_cosmos_db.py

```diff
@@ -111,72 +140,68 @@ def add_product(product_id, product_name, product_description, price):
     except exceptions.CosmosResourceExistsError:
         print(f"Product with product_id {product_id} already exists.")
 
-def preview_table(container_name):
-    database = client.get_database_client(database_name)
-    container = database.get_container_client(container_name)
-    items = container.query_items(
-        query="SELECT * FROM c",
-        enable_cross_partition_query=True
+
+def process_and_insert_data(
+        filename: str,
+        container: ContainerProxy,
+        vector_field: Optional[str] = None,
+        full_text_fields: Optional[List[str]] = None,
+):
+    if not os.path.exists(filename):
+        print(f"File {filename} not found.")
+        return
+
+    with open(filename, "r") as f:
+        data = json.load(f)
+
+    if len(data) > 300:
```

Comment:
I think this logic means that some records are skipped if number of json docs in the file is greater than 300? What is this for?